src/plugins/usage_collection/server/usage_counters/search/search.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  ISavedObjectsRepository,
+  SavedObjectsFindOptions,
+} from '@kbn/core-saved-objects-api-server';
+import {
+  serializeCounterKey,
+  type UsageCountersSavedObjectAttributes,
+  USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+} from '../saved_objects';
+import type {
+  UsageCounterSnapshot,
+  UsageCountersSearchOptions,
+  UsageCountersSearchParams,
+  UsageCountersSearchResult,
+} from '../types';
+import { usageCountersSearchParamsToKueryFilter } from './search_utils';
+
+export async function searchUsageCounters(
+  repository: ISavedObjectsRepository,
+  params: UsageCountersSearchParams,
+  options: UsageCountersSearchOptions = {}
+): Promise<UsageCountersSearchResult> {
+  const { namespace: filterNamespace } = params;
+  const { perPage = 100, page = 1 } = options;
+
+  const filter = usageCountersSearchParamsToKueryFilter(params);
+
+  const findParams: SavedObjectsFindOptions = {
+    ...(filterNamespace && { namespaces: [filterNamespace] }),
+    type: USAGE_COUNTERS_SAVED_OBJECT_TYPE,
+    sortField: 'updated_at',
+    sortOrder: 'desc',
+    filter,
+    searchFields: [USAGE_COUNTERS_SAVED_OBJECT_TYPE],
+    perPage,
+    page,
+  };
+  const res = await repository.find<UsageCountersSavedObjectAttributes>(findParams);
```

Comment:
should we use PIT search?

Reply:
Totally, added with https://github.com/elastic/kibana/pull/187665#discussion_r1684950657